ignite/templates/app/files-consumer/app/ibc.go.plush

```diff
@@ -0,0 +1,241 @@
+		app.IBCKeeper.PortKeeper,
+		app.AccountKeeper,
+		scopedICAHostKeeper,
+		app.MsgServiceRouter(),
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+	)
+	app.ICAControllerKeeper = icacontrollerkeeper.NewKeeper(
+		app.appCodec,
+		app.GetKey(icacontrollertypes.StoreKey),
+		app.GetSubspace(icacontrollertypes.SubModuleName),
+		app.IBCFeeKeeper, // use ics29 fee as ics4Wrapper in middleware stack
+		app.IBCKeeper.ChannelKeeper,
+		app.IBCKeeper.PortKeeper,
+		scopedICAControllerKeeper,
+		app.MsgServiceRouter(),
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+	)
+
+	// Create IBC modules with ibcfee middleware
+	transferIBCModule := ibcfee.NewIBCMiddleware(ibctransfer.NewIBCModule(app.TransferKeeper), app.IBCFeeKeeper)
+
+	// integration point for custom authentication modules
+	var noAuthzModule porttypes.IBCModule
+	icaControllerIBCModule := ibcfee.NewIBCMiddleware(
+		icacontroller.NewIBCMiddleware(noAuthzModule, app.ICAControllerKeeper),
+		app.IBCFeeKeeper,
+	)
+
+	icaHostIBCModule := ibcfee.NewIBCMiddleware(icahost.NewIBCModule(app.ICAHostKeeper), app.IBCFeeKeeper)
+
+	// Create static IBC router, add transfer route, then set and seal it
+	ibcRouter := porttypes.NewRouter().
+		AddRoute(ibctransfertypes.ModuleName, transferIBCModule).
+		AddRoute(icacontrollertypes.SubModuleName, icaControllerIBCModule).
+		AddRoute(ibcccvtypes.ModuleName, consumerModule).
+		AddRoute(icahosttypes.SubModuleName, icaHostIBCModule)
+
+	// this line is used by starport scaffolding # ibc/app/module
+
+	app.IBCKeeper.SetRouter(ibcRouter)
+
+	app.ScopedIBCKeeper = scopedIBCKeeper
+	app.ScopedIBCTransferKeeper = scopedIBCTransferKeeper
+	app.ScopedICAHostKeeper = scopedICAHostKeeper
+	app.ScopedICAControllerKeeper = scopedICAControllerKeeper
+
+	// register IBC modules
+	if err := app.RegisterModules(
+		ibc.NewAppModule(app.IBCKeeper),
+		ibctransfer.NewAppModule(app.TransferKeeper),
+		ibcfee.NewAppModule(app.IBCFeeKeeper),
+		icamodule.NewAppModule(&app.ICAControllerKeeper, &app.ICAHostKeeper),
+		capability.NewAppModule(app.appCodec, *app.CapabilityKeeper, false),
+		ibctm.NewAppModule(),
+		solomachine.NewAppModule(),
+		consumerModule,
+	); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Since the IBC modules don't support dependency injection, we need to
+// manually register the modules on the client side.
+// This needs to be removed after IBC supports App Wiring.
+func RegisterIBC(registry cdctypes.InterfaceRegistry) map[string]appmodule.AppModule {
+	modules := map[string]appmodule.AppModule{
+		ibcexported.ModuleName:      ibc.AppModule{},
+		ibctransfertypes.ModuleName: ibctransfer.AppModule{},
+		ibcfeetypes.ModuleName:      ibcfee.AppModule{},
+		icatypes.ModuleName:         icamodule.AppModule{},
+		capabilitytypes.ModuleName:  capability.AppModule{},
+		ibctm.ModuleName:            ibctm.AppModule{},
+		solomachine.ModuleName:      solomachine.AppModule{},
+		ibcconsumertypes.ModuleName: ibcconsumer.AppModule{},
```

Comment:
https://github.com/ignite/cli/pull/3660#discussion_r1516331951

Reply:
It's at the right place. Which module is missing?

Reply:
Sorry, I commented here to solve my last comment because I forgot to click the button